connectivity/FEATURE_BLE/source/generic/SecurityManagerImpl.cpp

```diff
@@ -205,7 +335,11 @@ ble_error_t SecurityManager::preserveBondingStateOnReset(bool enabled)
 
 ble_error_t SecurityManager::purgeAllBondingState(void)
 {
-    if (!_db) return BLE_ERROR_INITIALIZATION_INCOMPLETE;
+    tr_info("Purging all bonds");
+    if (!_db) {
+        tr_error("Failure, DB not initialized");
+        return BLE_ERROR_INITIALIZATION_INCOMPLETE;
+    }
```

Comment:
line 349 can fail without trace if _pal.set_csrk doesn't log traces

Reply:
But it will log trace, generaly, when the step is `error = operation; if (error) { return error; }` We can assume (in this context) that operation will trace the error and details of the call. 

I don't want to trace errors twice if it doesn't add meaningful information.

Reply:
yeah, as long as tracing is also added to pal